woocommerce/woocommerce-ios
# [HACK week] Move plugin list to SwiftUI and allow updating plugins in-app

Apologies for the 500+ PR, most of it comes from deleting the previous view controller. There's no need to review it this week, it can wait after HACK week has passed 🙇 

## Description
This PR updates the Plugin list view by moving it from UIKit to a SwiftUI. We've also updated plugin cells to use the existing `ProductRowView` component, which allows out-of-date plugins to be updated directly from the app (via a web view).

| Before | After |
|--------|--------|
| ![current plugin list](https://github.com/woocommerce/woocommerce-ios/assets/3812076/7e13c125-6a8f-4953-8e30-e8b45954b481) | ![Simulator Screenshot - iPhone 15 - 2024-03-13 at 12 38 17](https://github.com/woocommerce/woocommerce-ios/assets/3812076/10cca442-0dde-4a89-8258-45826333a44d) | 

## Changes
- We create a new `PluginListView`, which is mostly a scrollview for all the plugin rows.
- Since the Settings are mostly in UIKit, we present the new view through a `UIHostingController`
- Fetching from `StorageSitePlugin` does not have the data we're looking for, so we update the results controller to fetch `StorageSystemPlugin` instead.
- Removed existing view controller and related code.

## Testing instructions
1. Go to Menu > Settings > Plugins.
2. Observe that we have both `Up to Date` and `Update available` plugins (you can use https://indiemelon.mystagingwebsite.com for testing, there's a bunch of out of date plugins there)
3. Observe that tapping on `Up to Date` plugins does not trigger any action
4. Observe that tapping on `Update available` plugins will open a web view and require to authenticate (if needed)
5. In the wp-admin screen opened in the app browser, update any of the plugins, then close the webview. Observe how the plugin state has been updated in the list.

| Update available. Tap and login. | Updating via browser | Close browser. Up to date plugin. |
|--------|--------|--------|
| ![Simulator Screenshot - iPhone 15 - 2024-03-13 at 12 39 11](https://github.com/woocommerce/woocommerce-ios/assets/3812076/8fddc908-c545-4f9b-a22d-5cd0fe6a8e30) | ![Simulator Screenshot - iPhone 15 - 2024-03-13 at 12 39 41](https://github.com/woocommerce/woocommerce-ios/assets/3812076/00931d1e-e4c1-4fb1-ac2b-d6cdbb133cb0) | ![Simulator Screenshot - iPhone 15 - 2024-03-13 at 12 40 03](https://github.com/woocommerce/woocommerce-ios/assets/3812076/7c40c9ad-abf3-4218-bf80-277813fb0608) |
